brainstorm/src/components/items.tsx

```diff
@@ -78,15 +92,27 @@ export function ItemsView(props: {
 	}
 }
 
-/**
- * Removes a node from its parent {@link Items}.
- * If the note is not in an {@link Items}, it is left unchanged.
- */
-export function deleteItem(item: Item): void {
-	const parent = Tree.parent(item);
-	// Use type narrowing to ensure that parent is Items as expected for an Item.
-	if (Tree.is(parent, Items)) {
-		const index = parent.indexOf(item);
-		parent.removeAt(index);
+export interface ItemParent {
+	/**
+	 * Removes a child Item.
+	 */
+	deleteItem(item: Item): void;
+}
+
+interface HasItemParent extends TreeNode {
+	get [ItemParentSymbol](): ItemParent;
+}
+
+function tryAsItemParent(node: TreeNode): ItemParent | undefined {
+	const parent = Tree.parent(node);
```

Comment:
I don't think we want the call to Tree.parent inside this. I think the caller should do it. We may have cases where the calling code has a node, not the child item. Also this parent access is something the caller might need to deal with inval for, so putting it in here could hide inval issues